src/context.ts

```diff
@@ -2254,6 +2288,22 @@ type InlineQueryContextCore = FilterCore<"inline_query">;
  */
 export type InlineQueryContext<C extends Context> = Filter<C, "inline_query">;
 
+type ChosenInlineResultContextCore = FilterCore<"chosen_inline_result">;
+/**
+ * Type of the context object that is available inside the handlers for
+ * `bot.chosenInlineResult`.
+ *
+ * This helper type can be used to narrow down context objects the same way how
+ * annotate `bot.chosenInlineResult` does it. This allows you to context objects in
+ * middleware that is not directly passed to `bot.chosenInlineResult`, hence not
+ * inferring the correct type automatically. That way, handlers can be defined
```

Comment:
This is not relevant to this specific PR as other doc comments also include this, but I think its easier to read then. Maybe it should be improved in another PR?

```suggestion
 * annotate `bot.chosenInlineResult` does it. This allows you to give context to objects in
 * middleware that is not directly passed to `bot.chosenInlineResult`, hence not
 * inferring the correct type automatically. That way, handlers can be defined
```

Reply:
I'll update the docs string in a later PQ, thanks for pointing this out :)